### PR TITLE
[tests-only][full-ci]Added test for `space` public link

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR adds API tests scenarios for public link of space from viewer role perspective

Added Scenarios:
`public tries to upload resource in the public space`
`public tries to create resource in the public space`
`public tries to delete a file in the public space`
`public tries to delete a folder in the public space`
`public tries to change content of a resource in the public space`

### Related Issue:
https://github.com/owncloud/ocis/issues/4937

### CODE REVIEW

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 1]
```diff
@@ -0,0 +1,37 @@
+@api @skipOnOcV10
+Feature: A a viewer (public)
+  I should not be able to create/delete/upload/change resources
+
+
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:5]: Background won't accept tags
```suggestion
```

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+    And using spaces DAV path
+    And the administrator has given "Alice" the role "Space Admin" using the settings api
+    And user "Alice" has created a space "public space" with the default quota using the GraphApi
+    And user "Alice" has created a public link share of the space "public space" with settings:
+      | permissions | 1                        |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:14]:
```suggestion
      | permissions | 1 |
```

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+  Scenario: public tries to upload resource in the public space
+    When the public uploads file "test.txt" with content "test" using the new public WebDAV API
+    And the HTTP status code should be "403"
+
+
+  Scenario: public tries to create resource in the public space
+    When the public creates folder "created-by-public" using the new public WebDAV API
+    When the public creates folder "created-by-public1" using the new public WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:24]: need both steps?
```suggestion
    When the public creates folder "created-by-public" using the new public WebDAV API
```

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    And user "Alice" has created a public link share of the space "public space" with settings:
+      | permissions | 1                        |
+
+
+  Scenario: public tries to upload resource in the public space
+    When the public uploads file "test.txt" with content "test" using the new public WebDAV API
+    And the HTTP status code should be "403"
+
+
+  Scenario: public tries to create resource in the public space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:22]:
```suggestion
  Scenario: public tries to create folder in the public space
```

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    When the public creates folder "created-by-public" using the new public WebDAV API
+    When the public creates folder "created-by-public1" using the new public WebDAV API
+    And the HTTP status code should be "403"
+
+
+  Scenario: public tries to create resource in the public space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:28]:
```suggestion
  Scenario: public tries to delete resources in the public space
```

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      | username |
+      | Alice    |
+    And using spaces DAV path
+    And the administrator has given "Alice" the role "Space Admin" using the settings api
+    And user "Alice" has created a space "public space" with the default quota using the GraphApi
+    And user "Alice" has created a public link share of the space "public space" with settings:
+      | permissions | 1                        |
+
+
+  Scenario: public tries to upload resource in the public space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:17]:
```suggestion
  Scenario: public tries to upload resources in the public space
```

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    And the HTTP status code should be "403"
+
+
+  Scenario: public tries to create resource in the public space
+    Given user "Alice" has uploaded a file inside space "public space" with content "some content" to "test.txt"
+    When the public deletes file "test.txt" from the last public link share using the new public WebDAV API
+    And the HTTP status code should be "403"
+
+
+  Scenario: public tries to change resource in the public space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:34]:
```suggestion
  Scenario: public tries to change resources in the public space
```

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+      | Alice    |
+    And using spaces DAV path
+    And the administrator has given "Alice" the role "Space Admin" using the settings api
+    And user "Alice" has created a space "public space" with the default quota using the GraphApi
+    And user "Alice" has created a public link share of the space "public space" with settings:
+      | permissions | 1 |
+
+
+  Scenario: public tries to upload resources in the public space
+    When the public uploads file "test.txt" with content "test" using the new public WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:17]: Can we add the step for folder upload as well?

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    And the HTTP status code should be "403"
+
+
+  Scenario: public tries to create resources in the public space
+    When the public creates folder "created-by-public" using the new public WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:22]: Can we add the step for file creation as well?

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    And the HTTP status code should be "403"
+
+
+  Scenario: public tries to delete resources in the public space
+    Given user "Alice" has uploaded a file inside space "public space" with content "some content" to "test.txt"
+    When the public deletes file "test.txt" from the last public link share using the new public WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:28]: Can we add the step for folder deletion as well?

File: tests/acceptance/features/apiSpaces/publicLink.feature [hunk 11]
```diff
@@ -0,0 +1,35 @@
+@api @skipOnOcV10
+Feature: As a viewer (public)
+  I should not be able to create/delete/upload/change resources
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/publicLink.feature:3]: will this feature be used only for Viewer role? Also, you can add better feature description :smile:

**Review by saw-jan — COMMENTED** [review]
LGTM :+1:

**Review by individual-it — APPROVED** [review]

**Pull request merged**